### PR TITLE
feat(esp_bsp_generic): Support I2C Driver-NG

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
- Support I2C driver-NG for esp_bsp_generic and esp_bsp_devkit

### CODE REVIEW

**Comment by tore-espressif** [comment]
You will have to add
``` yml
examples/generic_button_led:
  disable:
    - if: IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 3
      reason: Requires I2C Driver-NG which was introduced in v5.2
```

to .build-test-rules.yml to fix the CI

File: bsp/esp_bsp_devkit/include/bsp/esp_bsp_devkit.h [hunk 1]
```diff
@@ -13,7 +13,8 @@
 
 #include "sdkconfig.h"
 #include "driver/gpio.h"
-#include "driver/i2c.h"
+#include "driver/i2s_std.h"
```
Comment (tore-espressif) [inline bsp/esp_bsp_devkit/include/bsp/esp_bsp_devkit.h:16]: Why is I2S added here?

File: bsp/esp_bsp_generic/include/bsp/esp_bsp_generic.h [hunk 2]
```diff
@@ -13,7 +13,8 @@
 
 #include "sdkconfig.h"
 #include "driver/gpio.h"
-#include "driver/i2c.h"
+#include "driver/i2s_std.h"
```
Comment (tore-espressif) [inline bsp/esp_bsp_generic/include/bsp/esp_bsp_generic.h:16]: Why was I2S added here?

File: .build-test-rules.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ examples:
   disable:
     - if: CONFIG_NAME in ["esp-box", "esp-box-lite"]
       reason: Do not build examples for deprecated BSPs
-    - if: (IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 3) and CONFIG_NAME in ["esp32_p4_function_ev_board", "esp-box-3"]
+    - if: (IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 3) and CONFIG_NAME in ["esp32_p4_function_ev_board", "esp-box-3", "esp_bsp_generic", "esp_bsp_devkit"]
       reason: Example depends on BSP, which is supported only for IDF >= 5.3
```
Comment (igrr) [inline .build-test-rules.yml:7]: esp_bsp_devkit's dependency is `  idf: ">=5.2"`, seems there is some inconsistency

File: bsp/esp_bsp_devkit/include/bsp/esp_bsp_devkit.h (hunk 1 above)
Comment (espzav) [reply bsp/esp_bsp_devkit/include/bsp/esp_bsp_devkit.h]: Copy-paste issue.

File: .build-test-rules.yml (hunk 3 above)
Comment (espzav) [reply .build-test-rules.yml:9]: Fixed, it should be supported from IDF 5.2

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**